CONTRIBUTING.md

```diff
@@ -37,11 +37,40 @@ We specifically do not merge PRs **without prior issues** that:
 Eager to get started on coding? Here's the short version:
 
 1. Set up a Go development environment with Git.
-2. Pay attention to copyright: [please read the DCO](https://developercertificate.org/), write the code yourself, avoid copy/paste. Disable or limit your AI coding assistant.
+2. Pay attention to copyright: [please read the DCO](https://developercertificate.org/), write the code yourself, avoid copy/paste. Disable your AI coding assistant.
 3. Run the tests with `go test` in the package you are working on.
 4. Build OpenTofu by running `go build ./cmd/tofu`.
 5. Update [the changelog](CHANGELOG.md).
 6. When you commit, use `git commit -s` to sign off your commits.
+7. Complete the checklist below before you submit your PR (or submit a draft PR).
+8. Your PR will be reviewed by the core team once it is marked as ready to review.
+
+---
+
+## PR checklist
+
+<!-- Make sure to keep this in sync with the PR template. -->
+
+Please make sure you complete the following checklist before you mark your PR ready for review. If you cannot complete the checklist but want to submit a PR, please submit it as a draft PR. Please note, the core team will only review your PR if you have completed the checklist and marked your PR as ready to review.
+
+- [ ] I have read the these contribution guidelines.
+- [ ] I have not used an AI coding assistant to create this PR.
+- [ ] I have marked any code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.
+
+### Go checklist
+
+If your PR contains Go code, please make sure you check off all items on this list:
+
+- [ ] I have run golangci-lint on my change and receive no errors relevant to my code.
+- [ ] I have added tests for all relevant use cases of my code.
```

Comment:
I think we should explicitly mention that existing tests must pass successfully (here and in the PR template):

```
- [ ] I have run existing tests and all passed successfully.
```